client/src/main/python/slipstream/HttpClient.py

```diff
@@ -18,272 +18,368 @@
 
 import os
 import time
-import base64
 import httplib
-import httplib2
+import requests
+import re
 import socket
-
+import stat
 from random import random
 from threading import Lock
+from urlparse import urlparse
+from cookielib import CookieJar
+from requests import Request
+from requests.cookies import MockResponse, MockRequest, RequestsCookieJar
+from six.moves.http_cookiejar import MozillaCookieJar
 
 import slipstream.exceptions.Exceptions as Exceptions
 import slipstream.util as util
 
 etree = util.importETree()
 
 
-class HttpClient(object):
+class SessionStore(requests.Session):
+    """Session with extended MozillaCookieJar file-based store.
+    """
+
+    def __init__(self, cookie_file=None):
+        super(SessionStore, self).__init__()
+        if cookie_file is None:
+            cookie_file = util.DEFAULT_COOKIE_FILE
+        cookie_dir = os.path.dirname(cookie_file)
+        self.cookies = MozillaCookieJar(cookie_file)
+        if not os.path.isdir(cookie_dir):
+            os.mkdir(cookie_dir, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+        if os.path.isfile(cookie_file):
+            self.cookies.load(ignore_discard=True)
+            self.cookies.clear_expired_cookies()
+
+    def request(self, *args, **kwargs):
+        response = super(SessionStore, self).request(*args, **kwargs)
+        if not self.verify and response.cookies:
+            self._unsecure_cookie(args[1], response)
+        self.cookies.save(ignore_discard=True)
+        return response
+
+    def _unsecure_cookie(self, url_str, response):
+        url = urlparse(url_str)
+        if url.scheme == 'http':
```

Comment:
because url_str is an argument, it might be safer to have `url.scheme.lower()` instead